pkg/cloudmap/cache.go

```diff
@@ -0,0 +1,148 @@
+package cloudmap
+
+import (
+	"fmt"
+	"github.com/aws/aws-cloud-map-mcs-controller-for-k8s/pkg/model"
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/util/cache"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"time"
+)
+
+const (
+	nsKeyPrefix    = "ns"
+	svcKeyPrefix   = "svc"
+	endptKeyPrefix = "endpt"
+
+	defaultCacheSize = 1024
```

Comment:
I would increase the default cache size. Considering all endpoints across all services need to be cached, it may easily grow above `1024` and items may get randomly evicted.

Reply:
Since we’re caching endpoints as list, it’s one entry per service, not endpoint.

Reply:
I see, I missed that. Then it's probably safe from the size perspective. Thanks.